# Ex2 - Getting and Knowing your Data

This time we are going to pull data directly from the internet.
Special thanks to: https://github.com/justmarkham for sharing the dataset and materials.

### Step 1. Import the necessary libraries

In [103]:
import pandas as pd

### Step 2. Import the dataset from this [address](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv). 

### Step 3. Assign it to a variable called chipo.

In [104]:
chipo= pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv", sep="\t")

### Step 4. See the first 10 entries

In [105]:
print(chipo.head(10).to_string())

   order_id  quantity                              item_name                                                                                                  choice_description item_price
0         1         1           Chips and Fresh Tomato Salsa                                                                                                                 NaN     $2.39 
1         1         1                                   Izze                                                                                                        [Clementine]     $3.39 
2         1         1                       Nantucket Nectar                                                                                                             [Apple]     $3.39 
3         1         1  Chips and Tomatillo-Green Chili Salsa                                                                                                                 NaN     $2.39 
4         2         2                           Chicken Bowl

### Step 5. What is the number of observations in the dataset?

In [106]:
# Solution 1
print(chipo.shape[1])

5


In [107]:
# Solution 2
print(chipo.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB
None


### Step 6. What is the number of columns in the dataset?

In [108]:
print(chipo.shape[0])

4622


### Step 7. Print the name of all the columns.

In [109]:
print(chipo.columns)

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')


### Step 8. How is the dataset indexed?

In [110]:
print(chipo.index)

RangeIndex(start=0, stop=4622, step=1)


### Step 9. Which was the most-ordered item? 

In [111]:
c = chipo.groupby('item_name')
c = c.sum() 
c.sort_values(by = ['quantity'], ascending=False, inplace=True)
c.head(1)

order_id  quantity  \
item_name                          
Chicken Bowl    713926       761   

                                             choice_description  \
item_name                                                         
Chicken Bowl  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...   

                                                     item_price  
item_name                                                        
Chicken Bowl  $16.98 $10.98 $11.25 $8.75 $8.49 $11.25 $8.75 ...

Most Ordered Item : Chicken Bowl

### Step 10. For the most-ordered item, how many items were ordered?

In [112]:
c = chipo.groupby('item_name')
c = c.sum()
c.sort_values(by = ['quantity'], ascending=False, inplace= True)
c.head(1)

order_id  quantity  \
item_name                          
Chicken Bowl    713926       761   

                                             choice_description  \
item_name                                                         
Chicken Bowl  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...   

                                                     item_price  
item_name                                                        
Chicken Bowl  $16.98 $10.98 $11.25 $8.75 $8.49 $11.25 $8.75 ...

761 items were ordered for Chicken bowl.

### Step 11. What was the most ordered item in the choice_description column?

In [113]:
c = chipo.groupby('choice_description')
c = c.sum()
c.sort_values(by = ['quantity'], ascending= False, inplace= True)
c.head(1)

order_id  quantity  \
choice_description                       
[Diet Coke]           123455       159   

                                                            item_name  \
choice_description                                                      
[Diet Coke]         Canned SodaCanned SodaCanned Soda6 Pack Soft D...   

                                                           item_price  
choice_description                                                     
[Diet Coke]         $2.18 $1.09 $1.09 $6.49 $2.18 $1.25 $1.09 $6.4...

### Step 12. How many items were orderd in total?

In [114]:
total_items_ordered = chipo.quantity.sum()
print(total_items_ordered)

4972


### Step 13. Turn the item price into a float

#### Step 13.a. Check the item price type

In [115]:
print(chipo.item_price.dtypes)

object


#### Step 13.b. Create a lambda function and change the type of item price

In [ ]:
chipo['item_price'] = chipo['item_price'].str[1:]
chipo['item_price'] = chipo['item_price'].apply(lambda x :float(x))

float64


#### Step 13.c. Check the item price type

In [119]:
print(chipo.item_price.dtype)

float64


### Step 14. How much was the revenue for the period in the dataset?

In [120]:
total_revenue=(chipo['item_price'] * chipo['quantity']).sum()
print(f"Total Revenue for the period is: ${total_revenue}")

Total Revenue for the period is: $39237.02


### Step 15. How many orders were made in the period?

In [122]:
total_orders = chipo['quantity'].sum()
print(f"Total Orders made in the period are: {total_orders}")

Total Orders made in the period are: 4972


### Step 16. What is the average revenue amount per order?

In [129]:
# Solution 1
chipo['revenue'] = chipo['quantity'] * chipo['item_price']
c = chipo.groupby('order_id').sum()
print(c['revenue'].mean().__round__(2))

21.39


In [133]:
# Solution 2
print(chipo.groupby('order_id').sum().revenue.mean().__round__(2))

21.39


### Step 17. How many different items are sold?

In [134]:
print(chipo.item_name.value_counts().count())

50
